app/services/ccms/add_case_service.rb

```diff
@@ -0,0 +1,26 @@
+module CCMS
+  class AddCaseService < BaseSubmissionService
+    def call
+      response = case_add_requestor.call
+      if case_add_response_parser(response).success?
+        submission.case_add_transaction_id = case_add_requestor.transaction_request_id
+        create_history(submission.aasm_state, :case_submitted)
+        submission.submit_case!
+      else
+        handle_failure(response)
+      end
+    rescue CcmsError, StandardError => e # TODO: Replace `StandardError` with list of known expected errors
+      handle_failure(e)
+    end
+
+    private
+
+    def case_add_requestor
+      @case_add_requestor ||= CaseAddRequestor.new(submission.legal_aid_application.applicant)
+    end
+
+    def case_add_response_parser(response)
```

Comment:
There are two scenarios here:
1. The `response` will change and therefore needs to be passed in each time. In which case it shouldn't be cached with `@case_add_response_parser ||=`
2. The `response` is set once in which case it does not need to be passed into this method as it is already available to this instance method.

Reply:
It's scenario 2. I'll change it.